lib/src/number_of_processors/processors_web.dart

```diff
@@ -1 +1,24 @@
-int get numberOfProcessors => 1;
+import 'dart:html';
+
+/// Web implementation to get the number of logical processors (CPU cores) available for worker tasks.
+/// Uses the browser's `navigator.hardwareConcurrency` API to determine the count.
+///
+/// The implementation reserves one core for the main thread by returning
+/// `hardwareConcurrency - 1` when multiple cores are available. This ensures
+/// the main thread has dedicated resources for UI and other critical operations.
+///
+/// Returns:
+///   * Number of logical processors minus 1 if multiple cores are available
+///   * 1 if only a single core is available or if hardwareConcurrency is not supported
+///
+/// This matches the behavior of the IO implementation which also reserves one core
+/// for the main thread.
+int get numberOfProcessors {
+  // Check if multiple cores are available and subtract one for the main thread
+  if (window.navigator.hardwareConcurrency != null &&
```

Comment:
Small adjustment for bulky if statement and good to go mate
```dart
return max((window.navigator.hardwareConcurrency ?? 1) - 1, 1)
```

Reply:
Also I would like to suggest reduce comments boilerplate to short and simple statement like:
"Uses navigator.hardwareConcurrency api, returns n - 1 threads, where n is number of available threads and (-1) as a reserved for main thread to avoid skipping frames"
This changes is not necessary, but I would like to see them

Reply:
Thanks for the suggestions, @machinescream! I agree with both points and will make the changes as suggested. I'll simplify the comment and adjust the if statement accordingly. 👍

Reply:
I've made the suggested changes to refactor the if statement and simplify the comments. Let me know if there's anything else you'd like to adjust!